src/main/java/picard/illumina/NewIlluminaBasecallsConverter.java

```diff
@@ -0,0 +1,392 @@
+                                         final Class<CLUSTER_OUTPUT_RECORD> outputRecordClass,
+                                         final BclQualityEvaluationStrategy bclQualityEvaluationStrategy,
+                                         final boolean applyEamssFiltering, final boolean includeNonPfReads,
+                                         final boolean ignoreUnexpectedBarcodes, int maxNoCalls, int maxMismatches,
+                                         int minMismatchDelta, int minimumBaseQuality, MetricsFile<BarcodeMetric, Integer> metrics, File metricsFile) {
+
+        super(barcodeRecordWriterMap, maxReadsInRamPerTile, tmpDirs, codecPrototype, ignoreUnexpectedBarcodes,
+                demultiplex, outputRecordComparator, includeNonPfReads, bclQualityEvaluationStrategy,
+                outputRecordClass, numProcessors, new IlluminaDataProviderFactory(basecallsDir,
+                        barcodesDir, lane, readStructure, bclQualityEvaluationStrategy));
+        this.maxNoCalls = maxNoCalls;
+        this.maxMismatches = maxMismatches;
+        this.minMismatchDelta = minMismatchDelta;
+        this.minimumBaseQuality = minimumBaseQuality;
+        this.tiles = new ArrayList<>();
+        this.metrics = metrics;
+        this.metricsFile = metricsFile;
+        int numBarcodes = readStructure.sampleBarcodes.length();
+
+        barcodeRecordWriterMap.keySet().forEach(barcode -> {
+            if (barcode != null) {
+                int pos = 0;
+                String[] bcStrings = new String[numBarcodes];
+                for (int i = 0; i < numBarcodes; i++) {
+                    int endIndex = readStructure.sampleBarcodes.getDescriptorLengths()[i];
+                    bcStrings[i] = barcode.substring(pos, endIndex + pos);
+                    pos += endIndex;
+                }
+                this.barcodesMetrics.put(barcode, new BarcodeMetric(null, null, barcode, bcStrings));
+            }
+            barcodeWriterThreads.put(barcode, new ThreadPoolExecutorWithExceptions(1));
+        });
+
+        File laneDir = new File(basecallsDir, IlluminaFileUtil.longLaneStr(lane));
+
+        File[] cycleDirs = IOUtil.getFilesMatchingRegexp(laneDir, IlluminaFileUtil.CYCLE_SUBDIRECTORY_PATTERN);
+
+        //CBCLs
+        cbcls = new ArrayList<>();
+        Arrays.asList(cycleDirs)
+                .forEach(cycleDir -> cbcls.addAll(
+                        Arrays.asList(IOUtil.getFilesMatchingRegexp(
+                                cycleDir, "^" + IlluminaFileUtil.longLaneStr(lane) + "_(\\d{1,5}).cbcl$"))));
+
+        if (cbcls.size() == 0) {
+            throw new PicardException("No CBCL files found.");
+        }
+
+        IOUtil.assertFilesAreReadable(cbcls);
+
+        //locs
+        File locsFile = new File(basecallsDir.getParentFile(), "s.locs");
+        LocsFileReader locsFileReader = new LocsFileReader(locsFile);
```

Comment:
Seems like you'd want to put this in a try/catch to avoid not closing the reader on exception. E.g.
```
        try (LocsFileReader locsFileReader = new LocsFileReader(locsFile)) {
            while (locsFileReader.hasNext()) {
                locs.add(locsFileReader.next());
            }
        }
```